### ISSUE TITLE
make dns optional to be in the config

### ISSUE THREAD

**Issue by fsheidaiee** [2dust/v2rayNG#808]
Hi
in every manual config in v2rayng this part of the config is always there in the beginning of the string:
  "dns": {
    "hosts": {
      "domain:googleapis.cn": "googleapis.com"
    },
    "servers": [
      "1.1.1.1"
    ]
  },
every time I have to customize config and remove this part
please make this optional in setting
thanks

**Comment by if1y** [comment]
Why remove? Pls give your reasons.

**Comment by fsheidaiee** [comment]
> Why remove? Pls give your reasons.

When using socks 5 in v2rayng i get "outbound proxyman failed" in some sites and can't open some sites in android while these sites have no problem in v2rayn on windows
I checked both configs and the difference was these lines.
By the way i don't live in china and the dns host declares that it works for china maybe not my country.
"dns": {
"hosts": {
"domain:googleapis.cn": "googleapis.com"

**Comment by fsheidaiee** [comment]
> Why remove? Pls give your reasons.

I noticed this only works on proxy mode
when I use vpn mode, dns is implemented even with customized config.
the problem is when I use tor proxy. I set v2ray on socks5 to listen to tor and use tor as vpn with v2ray rules. but I get udp packets and those packets won't get direct connection. despite of setting rule for that website to pass directly.
I have to mention that I don't have problem with other proxies like adguard. the problem occurs only with tor in android. using tor and v2rayn on windows is also without bug.
thanks
